torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -231,8 +231,16 @@ class ProcessGroupNCCL : public ProcessGroup {
       TORCH_INTERNAL_ASSERT(
           cudaEvents_->size() == 1,
           "FutureNCCL only supports single-process single-device mode.");
+      for (const at::cuda::CUDAEvent& event : *cudaEvents_) {
+        TORCH_INTERNAL_ASSERT(event.isCreated());
+        TORCH_INTERNAL_ASSERT(event.device_index() == deviceIndex_);
```

Comment:
I might miss sth. Why all events are on the same device? For multi-input collective calls, aren't we going to get one event per device?

Reply:
At this stage there is in fact a single event, which is on deviceIndex_ (see the check just above). I had no real reason to use a `for` loop here rather than just check the first (and only) element of `cudaEvents_`. If you want I can change this.